packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -298,63 +306,128 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
 
-    final List<Widget> children = <Widget>[
-      if (widget.title != null || widget.content != null)
-        Flexible(
-          flex: 3,
-          child: _CupertinoAlertContentSection(
-            title: widget.title,
-            message: widget.content,
-            scrollController: _effectiveScrollController,
-            titlePadding: EdgeInsets.only(
-              left: _kDialogEdgePadding,
-              right: _kDialogEdgePadding,
-              bottom: widget.content == null ? _kDialogEdgePadding : 1.0,
-              top: _kDialogEdgePadding * effectiveTextScaleFactor,
-            ),
-            messagePadding: EdgeInsets.only(
-              left: _kDialogEdgePadding,
-              right: _kDialogEdgePadding,
-              bottom: _kDialogEdgePadding * effectiveTextScaleFactor,
-              top: widget.title == null ? _kDialogEdgePadding : 1.0,
-            ),
-            titleTextStyle: _kCupertinoDialogTitleStyle.copyWith(
-              color: CupertinoDynamicColor.resolve(CupertinoColors.label, context),
-            ),
-            messageTextStyle: _kCupertinoDialogContentStyle.copyWith(
-              color: CupertinoDynamicColor.resolve(CupertinoColors.label, context),
-            ),
-          ),
-        ),
-    ];
+    final Widget child = _CupertinoAlertContentSection(
+      title: widget.title,
+      message: widget.content,
+      scrollController: _effectiveScrollController,
+      titlePadding: EdgeInsets.only(
+        left: _kDialogEdgePadding,
+        right: _kDialogEdgePadding,
+        bottom: widget.content == null ? _kDialogEdgePadding : 1.0,
+        top: _kDialogEdgePadding * effectiveTextScaleFactor,
+      ),
+      messagePadding: EdgeInsets.only(
+        left: _kDialogEdgePadding,
+        right: _kDialogEdgePadding,
+        bottom: _kDialogEdgePadding * effectiveTextScaleFactor,
+        top: widget.title == null ? _kDialogEdgePadding : 1.0,
+      ),
+      titleTextStyle: _kCupertinoDialogTitleStyle.copyWith(
+        color: CupertinoDynamicColor.resolve(CupertinoColors.label, context),
+      ),
+      messageTextStyle: _kCupertinoDialogContentStyle.copyWith(
+        color: CupertinoDynamicColor.resolve(CupertinoColors.label, context),
+      ),
+    );
 
     return ColoredBox(
       color: CupertinoDynamicColor.resolve(_kDialogColor, context),
-      child: Column(
-        mainAxisSize: MainAxisSize.min,
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: children,
-      ),
+      child: child,
     );
   }
 
-  Widget _buildActions() {
-    Widget actionSection = const LimitedBox(
-      maxWidth: 0,
-      child: SizedBox(width: double.infinity, height: 0),
-    );
-    if (widget.actions.isNotEmpty) {
-      actionSection = _CupertinoAlertActionSection(
+  void _onPressedUpdate(int actionIndex, bool isPressed) {
+    if (isPressed) {
+      setState(() {
+        _pressedIndex = actionIndex;
+      });
+    } else {
```

Comment:
uber nit: `else if`?

Reply:
I think nested `if` is more clear here, since it's basically saying "in the releasing scenario, let's check the index".